core/lib/zksync_core/src/consistency_checker/mod.rs

```diff
@@ -209,18 +209,17 @@ impl ConsistencyChecker {
     const DEFAULT_SLEEP_INTERVAL: Duration = Duration::from_secs(5);
 
     pub fn new(
-        web3_url: &str,
+        l1_client: impl EthInterface,
```

Comment:
Nit: `impl SomeTrait` is a hidden generic, which means that this function has to be monomorphized, which comes with a compilation time penalty. At the same time, this function already accepts `Arc<dyn Trait>`, which introduces an inconsistency. Given that there is no real need for this function to be generic, it's probably be better to accept `Box<dyn EthInterface>` directly. (and if we are to add a wider context, though it's even more of FYI, in the notion of the WIP [framework](https://github.com/matter-labs/zksync-era/tree/main/core/node/node_framework) `EthInterface` represents [a shared resource](https://github.com/matter-labs/zksync-era/blob/main/core/node/node_framework/src/implementations/resources/eth_interface.rs), so it can be made `Arc`).

Not a big deal, however, so feel free to leave as is.